File: tests/fixtures/adversarial/wrapper_factory.py
"""Adversarial fixture: wrapper factories hide direct constructor calls."""

import os

from langchain_openai import ChatOpenAI

MODEL_NAME = os.getenv("MODEL_NAME", "gpt-4o-mini")


def build_llm(model: str = MODEL_NAME):
    return ChatOpenAI(model=model)


def make_primary():
    constructor = ChatOpenAI
    return constructor(model="gpt-4o")


primary = build_llm()
fallback = make_primary()

File: tests/fixtures/adversarial/wrapper_factory.ts
import OpenAI from "openai";

const DEFAULT_MODEL = process.env.APP_MODEL || "gpt-4o-mini";

export function createClient(modelName: string = DEFAULT_MODEL) {
  return new OpenAI({ model: modelName });
}

export const client = createClient();

const PINNED_MODEL = "gpt-4o-mini";
export const pinned = new OpenAI({ model: PINNED_MODEL });
